### ISSUE TITLE
Improve pohint for upload date in #wesnoth-lib (src/gui/dialogs/addon/manager.cpp)

### ISSUE THREAD

**Issue by demario-w** [wesnoth/wesnoth#8822]
### Describe the desired feature

The 2 date strings from the textdomain #wesnoth-lib have the pohint:

> TRANSLATORS: Month + day of month + year + 12-hour time, eg 'November 02 2021, 1:59 PM'. Format for your locale.
> msgid "%B %d %Y, %I:%M %p"

> TRANSLATORS: Month + day of month + year + 24-hour time, eg 'November 02 2021, 13:59'. Format for your locale.
> msgid "%B %d %Y, %H:%M"

More details should be given on how to format after localization. A link to a reference to the standard for formatting is necessary if the pohint can't cover all cases.

At least what is needed:
- how to use a non-capitalized month name
- how to use a month number
- invalid characters

The French translation is currently "%d-%B-%Y %H:%M" which is possibly incorrect (month should be a number) and doesn't work either (result shown is "mars 13 2024, 20:31")

**Comment by Pentarctagon** [comment]
Documentation for the format string is at https://www.boost.org/doc/libs/1_85_0/libs/locale/doc/html/group__manipulators.html#ga2cbed7992bd001c4094275a980ddf4b3

> how to use a month number

`%m`

> how to use a non-capitalized month name

There is no option for that. Seems to be determined internally by Boost Locale.

> invalid characters

That's just everything not listed in the above documentation.

**Comment by stevecotton** [comment]
> result shown is "mars 13 2024, 20:31"

Running a local build of 1.18.0+dev in fr_FR, the add-on manager and load game dialog are both putting the day before the month for me. It might be a buggy dialog somewhere, please would you say where in the UI you're seeing it month-first?

**Comment by demario-w** [comment]
> where in the UI

The add-on manager

![addon-manager_fr](https://github.com/wesnoth/wesnoth/assets/97009180/7b62724c-6693-4e72-a3c9-ddfd92a6c39e)

**Comment by stevecotton** [comment]
Seems to work for me. Exactly which version are you seeing this on?

**Comment by Wedge009** [comment]
Official 1.18.0 release:
![image](https://github.com/wesnoth/wesnoth/assets/13679322/38033787-c5b2-4028-8be2-cee718eeb47d)

1.18 branch @ 3dec7e6c78b shows the same.

**Comment by Pentarctagon** [comment]
Is it perhaps platform or Boost version dependent? I see the same as Wedge009 does using the 1.18.0 Steam version on Linux.

**Comment by stevecotton** [comment]
Although the pohint has been added, the fix doesn't address the second issue about a "month day year" format appearing on @demario-w 's PC, but if we can't replicate it then I don't see a way to proceed with that.

@demario-w is there any chance that you'd set the fuzzy flag on the format string when testing that?

**Comment by demario-w** [comment]
> @demario-w is there any chance that you'd set the fuzzy flag on the format string when testing that?

@stevecotton The situation was that I followed [your indications](https://forums.wesnoth.org/viewtopic.php?p=672739#p672739) on how to use po file for direct testing, which includes removing the content under translations/fr. When filling the report, I removed the translations/wesnoth-lib containing the fr.po file to quote the exact output, but it left me with no translation file. Hence the output with EN format containing FR translated words. Sorry for the mix-up.
